lms/djangoapps/instructor/tests/test_enrollment.py

```diff
@@ -494,6 +497,95 @@ def test_delete_submission_scores_attempts_children(self):
         self.assertEqual(unrelated_state['brains'], 'zombie')
 
 
+class TestStudentModuleGrading(SharedModuleStoreTestCase):
+    """
+    Tests the effects of student module manipulations
+    on student grades.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TestStudentModuleGrading, cls).setUpClass()
+        cls.course = CourseFactory.create()
+        cls.course_key = cls.course.location.course_key
+        cls.chapter = ItemFactory.create(
+            parent=cls.course,
+            category="chapter",
+            display_name="Test Chapter"
+        )
+        cls.sequence = ItemFactory.create(
+            parent=cls.chapter,
+            category='sequential',
+            display_name="Test Sequential 1",
+            graded=True
+        )
+        cls.vertical = ItemFactory.create(
+            parent=cls.sequence,
+            category='vertical',
+            display_name='Test Vertical 1'
+        )
+        cls.problem = ItemFactory.create(
+            parent=cls.vertical,
+            category="problem",
+            display_name="Test Problem"
+        )
+        cls.user = UserFactory()
+
+    def test_delete_student_state(self):
+        problem_location = self.problem.location
+        # Create a student module for the user
+        StudentModule.objects.create(
```

Comment:
As an integration-level test, it seems incorrect to create django models directly.  Instead, consider simply calling [answer_problem](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/tests/utils.py#L31).  This would also allow you to remove all the calls to the submissions API below.